eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -160,20 +161,26 @@ jobs:
   - script: |
       set -x
 
-      if [[ -z '${{ parameters.reuseBuildArtifactsFrom }}' && '${{ parameters.withPreviousSDK }}' != 'True' ]]; then
-        docker run --rm -v "$(sourcesPath):/vmr" -w /vmr ${{ parameters.container }} ./prep.sh
-      else
+      customPrepArgs=""
+      prepSdk=true
+      if [[ '${{ parameters.withPreviousSDK }}' == 'True' ]]; then
+        # Source-built artifacts are from CentOS 8 Stream. We want to download them without
+        # downloading portable versions from the internet.
+        customPrepArgs="${customPrepArgs} --no-sdk --no-bootstrap"
+        prepSdk=false
+      elif [[ -n '${{ parameters.reuseBuildArtifactsFrom }}' ]]; then
+        customPrepArgs="${customPrepArgs} --no-sdk --no-artifacts"
+        prepSdk=false
+      fi
+
+      docker run --rm -v "$(sourcesPath):/vmr" -w /vmr ${{ parameters.container }} ./prep.sh ${customPrepArgs}
+      
+      if [[ "$prepSdk" == "false" ]]; then
```

Comment:
Although there are no scenarios that require it today, it feels like the right order would be to initialize the SDK prior to running prep because prep can have a dependency on the SDK depending on the options used.